simul/monitor/stats_test.go

```diff
@@ -212,37 +213,44 @@ func TestStatsString(t *testing.T) {
 	}
 }
 
+// this function generates random number for this simulation
+func random(min, max int) int {
+	rand.Seed(time.Now().UTC().UnixNano())
+	return rand.Intn((max-min)+1) + min
+}
+
 func TestStructParsedWithString(t *testing.T) {
-	testRange := [3]int{1, 2, 3}
-	for testCount, _ := range testRange {
-		mon, _ := setupMonitorStringTest(t, testCount)
-		//mon := monStatValues.m
-		dm := &DummyCounterIO{0, 0}
-		// create the counter measure
-		cm := NewCounterIOMeasure("dummy", dm)
-		if cm.baseRx != dm.rvalue || cm.baseTx != dm.wvalue {
-			t.Logf("baseRx = %d vs rvalue = %d || baseTx = %d vs wvalue = %d", cm.baseRx, dm.rvalue, cm.baseTx, dm.wvalue)
-			t.Fatal("Tx() / Rx() not working ?")
+	testRange := [5]int{1, 2, 3, 4, 5}
```

Comment:
Before this test was covering 3 cases, now you are covering only one case, and each time the test is run it's a different case. This is not how unit tests work: they should cover the exact same set of cases on every run, so that you can compare code coverage stats from one run to the next.